Dockerfile

```diff
@@ -11,9 +11,10 @@ RUN make build
 
 FROM node:14-alpine3.11 AS solidity-builder
 WORKDIR /firefly/solidity_firefly
-ADD solidity_firefly .
+ADD solidity_firefly/package*.json .
 RUN npm install
 RUN npm config set user 0
+ADD solidity_firefly .
```

Comment:
I'm sure there's a good reason for what you did, but why do you only bring the `package.json` in first, and then everything else after that?

Reply:
It allows Docker to cache the results of `npm install` (which only depends on package.json). That way if you make a small tweak to a Solidity file, you don't have to wait for Docker to re-download all the npm packages before compiling it.

Hint: I was making a lot of small tweaks to Solidity files (:

Reply:
Ah yep. That totally makes sense. Good idea.